There are only 2 failing tests:
```
E                   AssertionError: Expected aggregate count for cases_aggregated_by_ethnicity not matching output: aggregate={'ethnicity_display': 'White British', 'ethnicities': 10} should be expected_ethnicity_data={'ethnicity_display': 'White British', 'ethnicities': 11}
E                   assert False
```
Looks to be a mismatch between the number of expected cases seeded in the tests with an ethnicity of White British, compared with the calculated value. 

I have not touched any of that code.

For this PR - I have suggested a new button, here disabled as eligibility criteria for E12 have not been confirmed:
<img width="781" alt="image" src="https://github.com/rcpch/rcpch-audit-engine/assets/1058647/769d108e-d1ca-421b-88cc-cfdddf8cb9ca">

Here criteria have been confirmed so the date field is enabled but the button remains disabled as no date has yet been entered:
<img width="779" alt="image" src="https://github.com/rcpch/rcpch-audit-engine/assets/1058647/9eca29f0-544a-4618-964d-84240d667775">

This changes either when the datapicker selects a date, or a date including year have been typed:
<img width="778" alt="image" src="https://github.com/rcpch/rcpch-audit-engine/assets/1058647/d3b7f9aa-c33d-456e-a00f-2a823f16fe49">

Once a date has been selected and the set button clicked, the confirmation dialog appears as before with the subsequent workflow which remains unchanged.

--- a/templates/base.html
+++ b/templates/base.html
@@ -70,7 +70,7 @@
 <script src="https://unpkg.com/hyperscript.org@0.9.7"></script>
   <script src="{% static 'js/jquery.min.js' %}"></script>
   <script src="{% static 'htmx_1.8.4/htmx.min.js' %}"></script>
-  <script src="{% static 'hyperscript_0.9.8/hyperscript.min.js' %}"></script>
+  <script src="{% static 'hyperscript_0.9.12/hyperscript.min.js' %}"></script>
   <script src="{% static 'js/semantic.min.js' %}"
     integrity=""
     crossorigin="anonymous" referrerpolicy="no-referrer"></script>

--- a/templates/epilepsy12/partials/page_elements/date_field.html
+++ b/templates/epilepsy12/partials/page_elements/date_field.html
@@ -37,6 +37,7 @@
             <div class='ui left icon input'>
                 <input
                     {% if not enabled %}disabled{% endif %}
+                    _="on keyup or change if my value remove @disabled from #{{input_date_field_name}}_id"
                     type="date"
                     name="{{input_date_field_name}}"
                     value="{{date_value|date:'Y-m-d'}}">
@@ -50,8 +51,9 @@
                 </input>
                 <i class="htmx-indicator spinner loading icon" id="spinner"></i>
                 <button 
+                    id= "{{input_date_field_name}}_id"
                     class="ui rcpch_light_blue button"
-                    {% if not enabled %}disabled{% endif %}
+                    {% if not enabled or date_value is None %}disabled{% endif %}
                     hx-headers='{"X-CSRFToken": "{{ csrf_token }}"}'
                     hx-swap="{{hx_swap}}"
                     name="{{input_date_field_name}}"

--- a/templates/epilepsy12/partials/page_elements/date_field.html
+++ b/templates/epilepsy12/partials/page_elements/date_field.html
@@ -51,6 +51,7 @@
                 <i class="htmx-indicator spinner loading icon" id="spinner"></i>
                 <button 
                     class="ui rcpch_light_blue button"
+                    {% if not enabled %}disabled{% endif %}
                     hx-headers='{"X-CSRFToken": "{{ csrf_token }}"}'
                     hx-swap="{{hx_swap}}"
                     name="{{input_date_field_name}}"

--- a/templates/epilepsy12/partials/page_elements/date_field.html
+++ b/templates/epilepsy12/partials/page_elements/date_field.html
@@ -37,6 +37,27 @@
             <div class='ui left icon input'>
                 <input
                     {% if not enabled %}disabled{% endif %}
+                    type="date"
+                    name="{{input_date_field_name}}"
+                    value="{{date_value|date:'Y-m-d'}}">
+                    {% if date_value is None or error_message %}
+                        <span class="episode-date" data-tooltip="This field is mandatory" data-inverted="" data-position="top left">
+                            <i class="rcpch_light_blue dot circle outline icon" ></i>
+                        </span>
+                    {% else %}
+                        <i class="rcpch_pink check circle outline icon"></i>
+                    {% endif %}
+                </input>
+                <i class="htmx-indicator spinner loading icon" id="spinner"></i>
+                <button 
+                    class="ui rcpch_light_blue button"
+                    hx-headers='{"X-CSRFToken": "{{ csrf_token }}"}'
+                    hx-swap="{{hx_swap}}"
+                    name="{{input_date_field_name}}"
+                    hx-post="{{hx_post}}"
+                    hx-target="{{hx_target}}"
+                    hx-trigger="click"
+                    hx-include="[name='{{input_date_field_name}}']"
                     {% if hx_confirm %}
                     _="on htmx:confirm(issueRequest)
                         halt the event
@@ -52,23 +73,7 @@
                           })
                         if result.isConfirmed issueRequest()"
                     {% endif %}
-                    type="date"
-                    hx-headers='{"X-CSRFToken": "{{ csrf_token }}"}'
-                    hx-swap="{{hx_swap}}"
-                    name="{{input_date_field_name}}"
-                    hx-post="{{hx_post}}"
-                    hx-target="{{hx_target}}"
-                    hx-trigger="{{hx_trigger}}"
-                    value="{{date_value|date:'Y-m-d'}}">
-                    {% if date_value is None or error_message %}
-                        <span class="episode-date" data-tooltip="This field is mandatory" data-inverted="" data-position="top left">
-                            <i class="rcpch_light_blue dot circle outline icon" ></i>
-                        </span>
-                    {% else %}
-                        <i class="rcpch_pink check circle outline icon"></i>
-                    {% endif %}
-                </input>
-                <i class="htmx-indicator spinner loading icon" id="spinner"></i>
+                >Set</button>
             </div>
 
             {% if error_message %}

